ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+ */
+function imap_renamemailbox($stream_id, string $old_name, string $new_name): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_deletemailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_subscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_unsubscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_append($stream_id, string $folder, string $message, string $options = '', string $internal_date = ''): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_ping($stream_id): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_base64(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_qprint(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_8bit(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_binary(string $text) {}
+
+
+function imap_utf8(string $mime_encoded_text): string {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_status($stream_id, string $mailbox, int $options) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_mailboxmsginfo($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_setflag_full($stream_id, string $sequence, string $flag, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_clearflag_full($stream_id, string $sequence, string $flag, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_sort($stream_id, int $criteria, int $reverse, int $options = 0, string $search_criteria ='', string $charset = '') {}
+
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_uid($stream_id, int $msg_no) {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_msgno($stream_id, int $unique_msg_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_list($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_lsub($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_getsubscribed($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_getmailboxes($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_fetch_overview($stream_id, string $sequence, int $options = 0) {}
+
+function imap_alerts(): array {}
+
+function imap_errors(): array {}
+
+/**
+ * @return string|false
+ */
+function imap_last_error() {}
+
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_search($stream_id, string $criteria, int $options = 0, string $charset = '') {}
```

Comment:
`$options` defaults to `SE_FREE` which is defined as 2 in my c-client version. Not sure if we should use `2` or `UNKNOWN`. ``$charset` is definitely `UNKNOWN`.

Reply:
SE_FREE is a defined user land constant for the imap ext, is that appropriate to use here?

Reply:
Ah, indeed! So it's appropriate to use it.